code/drasil-code/lib/Language/Drasil/Code/Imperative/WriteReadMe.hs

```diff
@@ -25,19 +25,21 @@ makeReadMe ReadMeInfo {
         caseName = name,
         examplePurpose = purp,
         exampleDescr = descr,
-        folderNum = number} = 
+        folderNum = number,
+        inputf = inName,
+        outputf = outName} = 
     makeMd [introInfo name auths $ fieldEmptySTR purp,
     whatInfo $ fieldEmptySTR descr,
-    makeInstr imptype configFPs name,
+    makeInstr imptype configFPs name inName outName,
     verInfo progLang progLangVers,
     unsupOS unsupportedOSs,
     extLibSec extLibns extLibfp,
     endNote number auths] -- add date information to end note for license
 
 -- | Helper for encoding the type of program (either library or controller-based) in a README file.
-makeInstr :: ImplementationType -> [FilePath] -> String -> Doc
-makeInstr Library _ = const empty
-makeInstr Program cfp = instDoc cfp
+makeInstr :: ImplementationType -> [FilePath] -> String  -> Maybe String -> Maybe String -> Doc
```

Comment:
5 argument functions are a kind of 'design smell' that says that the wrong data is being passed-in. Maybe it's not worth unbundling some of it?

Reply:
Sorry for my part in leading you astray! 😅😔